### ISSUE TITLE
Ideas for integration with exampletestr

### ISSUE THREAD

**Issue by rorynolan** [mikldk/roxytest#11]
Hello! I am the maintainer of `exampletestr` and I _love_ the new `@textexamples` tag!
I'm thinking of developing `exampletestr` to be able to help people to write tests for their examples in this way, rather than just through files in `tests/testthat/`. I might even make that the default behaviour (not sure but leaning that way). If I do this, it's probably an option that `roxytest` should also have (in my opinion). Would you be interested in wrapping this functionality into `roxytest` if I implement it?

**Comment by mikldk** [comment]
Interesting idea. I am not sure how you would obtain the desired result without automatically changing files in `R/`. I would like to see the implementation and functionality before committing to anything, but it does indeed sound interesting.

**Comment by rorynolan** [comment]
You're right. It would change the files in R/, but only the roxygen commented lines. It wouldn't change code or docs. exampletestr is already pretty invasive on that it creates files in tests/.
I would also have a one file at a time option whereby the new testexamples tag stuff would be copied to clipboard for the user to paste in. This would not change any files by itself. 
BTW don't feel the need to commit to anything here, I'm just seeking your ideas and making suggestions.

**Comment by mikldk** [comment]
I don't think it's invasive to generate files in `tests/` with the proper checks. But I think it is problematic to change user-created files in `R/`. Can it be implemented by new files in `R/`?

**Comment by rubak** [comment]
I think I need to understand `exampletestr` a bit better to give my opinion. Is it to be considered a bit like `package.skeleton` which generates boiler-plate code for the user that then has to fill in the details?

In that case, I guess the boiler-plate code really has to go into the original R files. I'm also a bit sceptical of automatically writing to these... I think a copy/paste solution (potentially helped by a RStudio addin) could be interesting to explore.

**Comment by rorynolan** [comment]
Yes, `package.skeleton()` is a good analogy, it is exactly _code for user to fill in the details_.
I've taken on board your hesitance to write directly to `R/` files and gone with the copy/paste solution, including an RStudio addin. This is all ready in the dev version (rorynolan/exampletestr) and you can read instructions on how to use the `roxytest` functionality at https://rorynolan.github.io/exampletestr/articles/one-function-at-a-time.html#roxytest-functionality.
Let me know what you think. I'll hold the next release of `exampletestr` to CRAN until `roxytest` is up there.
And thanks for your replies, I really appreciate the input.

**Comment by mikldk** [comment]
I haven't tried it out yet, but it looks like a useful addition. Thanks.
